network/mq/consumer.go

```diff
@@ -45,13 +56,16 @@ func (c *consumer) Reconnect(ctx context.Context) error {
 	return nil
 }
 
-func (c *consumer) consume(ctx context.Context, messages <-chan amqp.Delivery) {
+func (c *consumer) consume(ctx context.Context) {
 	for {
 		select {
 		case <-ctx.Done():
-			log.Infof("Stopped consuming queue %s", c.queue.Name())
+			log.Infof("Finished consuming queue %s", c.queue.Name())
+			return
+		case <-c.stopChan:
+			log.Infof("Force stopped consuming queue %s", c.queue.Name())
 			return
-		case msg := <-messages:
+		case msg := <-c.messages:
```

Comment:
As figured out, when mq connection is lost, this channel gets closed and therefore the `case` statement is used infinitely.